### DISCUSSION TITLE
Namespace Database\Migrations

### DISCUSSION THREAD

**Discussion by SoftwareDesign-Solution** [laravel/framework#34377] [General]
Seeders and factories have their own namespace. Why wasn't migrations also assigned to a namespace?

**Comment by driesvints** [comment]
Because the filenames aren't PSR-4 compatible

**Reply by str** [reply]
phpcs and idea complain because of the missing namespace for migration files, and my OCD complains seeing a red file in the IDE.   The namespace could be set even if the migration filenames are not PSR-4 compatible.  It would be better to generate the migration file names PSR-4 compatible, but it is not a requirement.
